libs/core/src/lib/select/commons/select-commons.ts

```diff
@@ -0,0 +1,994 @@
+* INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE
+* AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+* DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*
+*
+* */
+import { ActiveDescendantKeyManager, LiveAnnouncer } from '@angular/cdk/a11y';
+import { FdOptionSelectionChange, OptionComponent } from '../option/option.component';
+import { FdSelectChange } from '../select.component';
+import { ChangeDetectorRef, ElementRef, EventEmitter, isDevMode, NgZone, QueryList, Input } from '@angular/core';
+import { Directionality } from '@angular/cdk/bidi';
+import { defer, merge, Observable, Subject } from 'rxjs';
+import { CdkConnectedOverlay, ConnectedPosition, ViewportRuler } from '@angular/cdk/overlay';
+import { SelectionModel } from '@angular/cdk/collections';
+import { NgControl } from '@angular/forms';
+import { startWith, switchMap, take, takeUntil } from 'rxjs/operators';
+import { DOWN_ARROW, END, ENTER, hasModifierKey, HOME, SPACE, UP_ARROW } from '@angular/cdk/keycodes';
+
+
+ import { KeyUtil } from '../../utils/functions/key-util';
+
+/**
+ * Common behaviors from Material Select Component to build Fundamental-ngx Select on top of it. We need this solid
+ * foundation to deliver robust solution and only focus on additional Fiori 3 enhancement.
+ *
+ * Even this seems to be implemented as one big mixins with single anonymous class all this is going to be broking down
+ * into several pieces in order to maximize reusability. Such as a possible candidates would be to separate
+ * Keyboard Support, Overlay support and option panel positioning and maybe select behavior itself.
+ *
+ * @Hidden
+ */
+
+/**
+ * The select panel will only "fit" inside the viewport if it is positioned at
+ * this value or more away from the viewport boundary.
+ */
+export const SELECT_PANEL_VIEWPORT_PADDING = 5;
+
+/** The height of the select items in `em` units. */
+export const SELECT_ITEM_HEIGHT_EM = 3;
+
+/**
+ * Dropdown Header identifier. Since existing logic is relaying on the options[] for the height calculation of
+ * the panel existing native with LI does not fit into this structure.
+ *
+ * We need to improve this implementation to have something like fd-option-header that we can query and provide
+ * more readable structure instead of allowing user to mix <fd-option> with LI element on the same level.
+ *
+ * todo: This should temporary solution used only for current calculation until we introduce fd-option-header
+ */
+
+const SELECT_HEADER_IDENTIFIER = '.fd-list__group-header';
+
+export type Constructor<T> = new (...args: any[]) => T;
+
+/**
+ * Common Interface to define shareable properties and method
+ *
+ * @hidden
+ */
+export interface MatSelectCommonBehavior {
```

Comment:
I see values of bringing this into library, there are a lot of repeated functionalities, such as focus support / rewritten overlay(Popover) methods .

But there is integration with selection model and list key manager that IMO should be completely separated, wrapped inside reusable standalone services, so we can get rid of this big class.

Reply:
This implementation was commented already many times, we can always improve things, but now its how it is in material select - whcih reminds me @sKudum have you updated to latest code changes that are in version 10 for material select ?

Reply:
Hi @fkolar, I have compared latest select implementation with the changes we did in previous PR. 

 It is observed, that most of the changes done in past few months in material select are with respect to styles, directions and accessibility which are already taken care in popover component as per Fiori standards for us and tried  not to break already existing features in select given.

Considering above scenarios, In this PR I have used names for classes and properties as suggested in previous PR for mat fields and for styles, overlay, directions, positioning using popover component. For selectionModel, keyManager used material approach.

Please do let me know in case am missing anything in this process.